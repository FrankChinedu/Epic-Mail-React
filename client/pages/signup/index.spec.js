import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from './index';

describe('Sign up', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Signup />);

    expect(component).toMatchSnapshot();
  });
});
