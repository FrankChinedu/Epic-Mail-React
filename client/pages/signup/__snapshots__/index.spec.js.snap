// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up should render correctly in "debug" mode 1`] = `<signupComponent />`;

exports[`Sign up should render signup with store 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(Signup)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Signup
        authError={null}
        onSignUp={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        user={Array []}
      >
        <main
          className="flex justify-content-ctr mt-55"
        >
          <div
            className="col-5 mt-5 form-cover"
          >
            <h3
              className="center-text"
            >
              Sign Up
            </h3>
            <Formik
              enableReinitialize={false}
              initialValues={
                Object {
                  "email": "",
                  "firstname": "",
                  "lastname": "",
                  "password": "",
                  "passwordConfirmation": "",
                }
              }
              isInitialValid={false}
              onSubmit={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": undefined,
                  "_nodes": Array [
                    "passwordConfirmation",
                    "password",
                    "lastname",
                    "firstname",
                    "email",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "fields": Object {
                    "email": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": true,
                        "undefined": false,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "firstname": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "lastname": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "password": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "min": true,
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "passwordConfirmation": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {
                          null,
                        },
                        "refs": Map {
                          "password" => Reference {
                            "getter": [Function],
                            "isContext": false,
                            "isSibling": true,
                            "isValue": false,
                            "key": "password",
                            "map": undefined,
                            "path": "password",
                          },
                        },
                      },
                      "_whitelistError": [Function],
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="firstName"
                  >
                    First Name
                  </label>
                  <input
                    className="form-control mt-5"
                    name="firstname"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="lastname"
                  >
                    Last Name
                  </label>
                  <input
                    className="form-control mt-5"
                    name="lastname"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <input
                    className="form-control mt-5"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    autoComplete="true"
                    className="form-control mt-5"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="passwordConfirmation"
                  >
                    Confirm Password
                  </label>
                  <input
                    autoComplete="true"
                    className="form-control mt-5"
                    name="passwordConfirmation"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="center-text mt-25"
                />
                <div
                  className="mt-25 center-text"
                >
                  <button
                    className="mt-25 btn"
                    disabled={false}
                    type="submit"
                  >
                    Sign Up
                  </button>
                </div>
              </form>
            </Formik>
          </div>
        </main>
      </Signup>
    </Connect(Signup)>
  </Router>
</BrowserRouter>
`;
